onnxruntime/core/session/inference_session.cc

```diff
@@ -297,6 +297,14 @@ void InferenceSession::ConstructorCommon(const SessionOptions& session_options,
                              session_options_.execution_mode == ExecutionMode::ORT_SEQUENTIAL &&
                              to.affinity_vec_len == 0;
       to.allow_spinning = allow_intra_op_spinning;
+
+      // Set custom threading functions
+      to.custom_create_thread_fn = session_options_.custom_create_thread_fn;
+      to.custom_thread_creation_options = session_options.custom_thread_creation_options;
+      to.custom_join_thread_fn = session_options_.custom_join_thread_fn;
+      if (to.custom_create_thread_fn) {
+        ORT_ENFORCE(to.custom_join_thread_fn, "custom join thread function not set for intra op thread pool");
```

Comment:
If this function is mandatory for the user to pass it should be spelled out in the C API documentation.

Reply:
https://github.com/microsoft/onnxruntime/blob/9fe36e35da772ecc1c946d56389cacf555566409/include/onnxruntime/core/session/onnxruntime_c_api.h#L3098